merkle-tree/hash-set/src/lib.rs

```diff
@@ -431,60 +432,102 @@ where
         value: &BigUint,
         current_sequence_number: usize,
     ) -> Result<(), HashSetError> {
+        let index_bucket = self.find_element_iter(value, current_sequence_number, 0, 20)?;
```

Comment:
Where does the `20` value come from? In any case, it would be worth to define it as a global `const` with a comment.

Reply:
observation that with reasonable size iterations should stay well under 20.
At the same time we always check 20 values now so that we want to keep this number as small as possible because every additional comparison we do costs more CU.

Reply:
lets do in followup pr